src/item_location.cpp

```diff
@@ -505,6 +500,16 @@ item_location::operator bool() const
     return ptr->valid();
 }
 
+bool item_location::empty() const
+{
+    return !ptr->valid();
+}
+
+void item_location::clear()
```

Comment:
This is a very unusual behavior. Why are you not calling `item_location::remove_item` directly?